Redefinition errors in WinSock2.h and winsock.h

With msvc 2015 the following sample doesn't build

````
#include <visp3/visp_core.h>

int main() { }
````

The build produces the following errors:

````
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\shared\ws2def.h(103): warning C4005: 'AF_IPX': macro redefinition
1>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\um\winsock.h(457): note: see previous definition of 'AF_IPX'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\shared\ws2def.h(147): warning C4005: 'AF_MAX': macro redefinition
1>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\um\winsock.h(476): note: see previous definition of 'AF_MAX'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\shared\ws2def.h(185): warning C4005: 'SO_DONTLINGER': macro redefinition
1>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\um\winsock.h(399): note: see previous definition of 'SO_DONTLINGER'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\shared\ws2def.h(235): error C2011: 'sockaddr': 'struct' type redefinition
1>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.10586.0\um\winsock.h(1007): note: see declaration of 'sockaddr'
````

--- a/modules/core/include/visp3/core/vpImageConvert.h
+++ b/modules/core/include/visp3/core/vpImageConvert.h
@@ -78,6 +78,9 @@
 #endif
 
 #if defined(_WIN32)
+// Include WinSock2.h before windows.h to ensure that winsock.h is not included by windows.h 
+// since winsock.h and winsock2.h are incompatible
+#  include <WinSock2.h> 
 #  include <windows.h>
 #endif
 

--- a/modules/core/include/visp3/core/vpMutex.h
+++ b/modules/core/include/visp3/core/vpMutex.h
@@ -47,6 +47,9 @@
 #if defined(VISP_HAVE_PTHREAD)
 #  include <pthread.h>
 #elif defined(_WIN32)
+// Include WinSock2.h before windows.h to ensure that winsock.h is not included by windows.h 
+// since winsock.h and winsock2.h are incompatible
+#  include <WinSock2.h> 
 #  include <windows.h>
 #endif
 

--- a/modules/core/include/visp3/core/vpThread.h
+++ b/modules/core/include/visp3/core/vpThread.h
@@ -10,6 +10,9 @@
 #  include <pthread.h>
 #  include <string.h>
 #elif defined(_WIN32)
+// Include WinSock2.h before windows.h to ensure that winsock.h is not included by windows.h 
+// since winsock.h and winsock2.h are incompatible
+#  include <WinSock2.h> 
 #  include <windows.h>
 #endif
 

